Sources/Model/PhotoCollectionViewDataSource.swift

```diff
@@ -63,18 +63,23 @@ final class PhotoCollectionViewDataSource : NSObject, UICollectionViewDataSource
         if let settings = settings {
             cell.settings = settings
         }
-        
+
+        // Clear any previously shown image
+        cell.imageView.image = nil
+
         // Cancel any pending image requests
         if cell.tag != 0 {
-            photosManager.cancelImageRequest(PHImageRequestID(cell.tag))
+            photosManager.cancelImageRequest(PHImageRequestID(Int32(cell.tag)))
         }
         
         let asset = fetchResult[indexPath.row]
         cell.asset = asset
         
         // Request image
         cell.tag = Int(photosManager.requestImage(for: asset, targetSize: imageSize, contentMode: imageContentMode, options: imageRequestOptions) { (result, _) in
```

Comment:
This is called when a request is cancelled, and the image was being set to nil, looks like this was causing some cells to be 'blank', as well as the nil image request options.